compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -490,12 +512,15 @@ macro_rules! define_queries {
                     key: query_keys::$name<'tcx>,
                     mode: QueryMode,
                 ) -> Option<Erase<query_values::$name<'tcx>>> {
```

Comment:
You can now create a new function named `__rust_end_short_backtrace` which calls `$name` here. Then pass that function to the `QueryEngine` constructor below. That avoids an extra function call.

Reply:
I figured another solution to avoids an extra function call,
since we are only checking whether the function name contains "__rust_end_short_backtrace" here: 
https://github.com/rust-lang/rust/blob/d3edfd18c790971c77845bfc1a2be4f9281c5416/library/std/src/sys_common/backtrace.rs#L80

why not we just add `__rust_end_short_backtrace` at the starting of `query`, such as commit: https://github.com/rust-lang/rust/pull/108938/commits/b871ec8d55abfe08b5da503df16cff510e674d21

I verified it works in dev.

Reply:
It will be inlined so that won't work without debug info.